lang/cs/Org.Apache.REEF.Tests/Functional/Messaging/MessageTask.cs

```diff
@@ -59,44 +69,33 @@ public Optional<TaskMessage> Message
 
         public byte[] Call(byte[] memento)
         {
-            Console.WriteLine("Hello, CLR TaskMsg!");
-            Thread.Sleep(5 * 1000);
-            return null;
-        }
+            while (!(_messageManager.IsTaskMessageSent && _messageManager.IsContextMessageSent))
+            {
+                // Do not exit until task and context messages are both sent.
+                Thread.Sleep(5000);
+            }
 
-        public void Dispose()
-        {
-            LOGGER.Log(Level.Info, "TaskMsg disposed.");
+            return null;
         }
 
-        private void DriverMessage(string message)
+        public void Handle(IDriverMessage value)
         {
-            if (!message.Equals(MessageDriver.Message))
+            if (!value.Message.IsPresent())
             {
-                Exceptions.Throw(new Exception("Unexpected driver message: " + message), "Unexpected driver message received: " + message, LOGGER);
+                throw new Exception("Expecting message from Driver but got missing message.");
             }
-        }
 
-        public class MessagingDriverMessageHandler : IDriverMessageHandler
-        {
-            private readonly MessageTask _parentTask;
+            Logger.Log(Level.Info, MessageReceivedFromDriverLog);
 
-            [Inject]
-            public MessagingDriverMessageHandler(MessageTask task)
+            var message = ByteUtilities.ByteArraysToString(value.Message.Value);
+            if (!message.Equals(MessageDriver.Message))
             {
-                _parentTask = task;
+                Exceptions.Throw(new Exception("Unexpected driver message: " + message), "Unexpected driver message received: " + message, Logger);
             }
+        }
```

Comment:
This method does not log success. Is this on purpose? It seems the other way around in all the other methods.